Sources/MongoSwift/Operations/UpdateOperation.swift

```diff
@@ -0,0 +1,134 @@
+import mongoc
+
+internal protocol UpdateOperationOptions: Encodable {
```

Comment:
this protocol was necessary because `UpdateOne`/`UpdateMany` use a different options type from `ReplaceOne`.

Reply:
should we just call this `WriteConcernProvider` and reuse it elsewhere?